slice/slice.go

```diff
@@ -71,55 +73,16 @@ func Partition[T any](vs []T, keep func(T) bool) []T {
 // of identical elements.  It returns a prefix of vs that contains the first
 // element of each run found, in their original relative order.  It takes time
 // proportional to len(vs) and does not allocate storage outside the slice.
+// Dedup zeroes the elements between the new length and the original length.
 //
 // The returned slice will contain (non-consecutive) duplicates only if
 // vs is not in sorted order at input. If vs is sorted at input (in either
 // direction), the elements of the prefix are exactly the unique first elements
 // of the input.
+//
+// Deprecated: Use the equivalent [slices.Compact] instead.
```

Comment:
I think we could safely cut this comment down:
```go
// Dedup rearranges the elements of vs in-place to deduplicate consecutive
// runs of identical elements. It returns a prefix of vs that contains the
// first element of each run found.
//
// Deprecated: Use the equivalent [slices.Compact] instead.
```

And IMO we could get rid of the test as well.

Reply:
Docstring shortened, and test gubbins deleted.